crates/volta-core/src/tool/package/fetch.rs

```diff
@@ -79,9 +76,90 @@ fn load_cached_distro(file: &Path, shasum_file: &Path) -> Option<Box<dyn Archive
     Tarball::load(distro).ok()
 }
 
-fn fetch_remote_distro(spec: tool::Spec, url: &str, path: &Path) -> Fallible<Box<Archive>> {
-    debug!("Downloading {} from {}, to {}", &spec, &url, path.display());
-    Tarball::fetch(url, path).with_context(download_tool_error(spec, url.to_string()))
+fn fetch_remote_distro(
+    path: &Path,
+    name: &str,
+    details: &PackageDetails,
+    session: &mut Session,
+) -> Fallible<Box<Archive>> {
+    ensure_containing_dir_exists(&path).with_context(|_| ErrorDetails::ContainingDirError {
+        path: path.to_path_buf(),
+    })?;
+
+    let dir = path.parent().unwrap();
+
+    let command = npm_pack_command_for(name, &details.version.to_string()[..], session, Some(dir))?;
+    debug!("Running command: `{:?}`", command);
+
+    let spinner = progress_spinner(&format!(
+        "Downloading via {} npm pack to {}",
```

Comment:
Small wording change: On the spinner, we shouldn't show the download directory or the fact that we use `npm pack`, that's an implementation detail that doesn't matter to the user (though we should still have a `debug!` message showing where it is downloading to). Here, we should just show that we are downloading the tool version:

```suggestion
        "Downloading {}",
```